muskathlon/reports/muskathlon_report.py

```diff
@@ -0,0 +1,119 @@
+        # pair and account invoice IDs are impair.
+        # many fields are hardcoded due to csv exportation needs
+        tools.drop_view_if_exists(self.env.cr, self._table)
+        self.env.cr.execute("""
+            CREATE OR REPLACE VIEW {table} AS (
+              SELECT
+                (2 * ROW_NUMBER() OVER (ORDER BY (SELECT 100))) AS id,
+                rc.id AS rc_id,
+                rc.partner_id,
+                rc.user_id,
+                1000 AS amount,
+                100000 AS amount_cent,
+                rc.sent_to_4m,
+                rc.payment_mode_id,
+                rco.event_id,
+                NULL AS journal_id,
+                'CHF' AS currency,
+                rc.muskathlon_registration_id,
+                rp.muskathlon_participant_id,
+                rp.name AS sponsorship_name,
+                rc.start_date AS date,
+                'success' AS status,
+                'sponsor' AS type,
+                'transfer' AS payment_methode,
+                rco.event_id AS project_id
+              FROM recurring_contract AS rc
+              LEFT JOIN res_partner AS rp ON rp.id = rc.partner_id
+              LEFT JOIN recurring_contract_origin AS rco
+                ON rc.origin_id = rco.id
+              LEFT JOIN crm_event_compassion AS cec ON rco.event_id = cec.id
+              WHERE cec.muskathlon_event_id IS NOT NULL
+            )
+            UNION ALL (
+              SELECT
+                (2 * ROW_NUMBER() OVER (ORDER BY (SELECT 100)) - 1) AS id,
+                ail.id AS ail_id,
+                ail.partner_id,
+                ail.user_id,
+                ail.price_subtotal,
+                ail.price_subtotal * 100,
+                ail.sent_to_4m,
+                ai.payment_mode_id,
+                ail.event_id,
+                aml.journal_id
+                AS journal_id,
+                rcu.name as currency,
+                ail.muskathlon_registration_id,
+                rp.muskathlon_participant_id,
+                rp.name AS sponsorship_name,
+                ai.date_invoice AS date,
+                'success' AS status,
+                'sponsor' AS type,
+                'transfer' AS payment_methode,
+                ail.event_id AS project_id
+              FROM account_invoice_line AS ail
+              LEFT JOIN account_invoice AS ai ON ail.invoice_id = ai.id
+              LEFT JOIN res_partner AS rp ON rp.id = ai.partner_id
+              LEFT JOIN res_currency AS rcu ON rcu.id = ai.currency_id
+              LEFT JOIN crm_event_compassion AS cec ON ail.event_id = cec.id
+              LEFT JOIN account_invoice_account_move_line_rel AS aiamlr
+                ON ai.id = aiamlr.account_invoice_id
+              LEFT JOIN account_move_line AS aml
+                ON aml.id = aiamlr.account_move_line_id
+              WHERE ail.state = 'paid' AND cec.muskathlon_event_id IS NOT NULL
+            )
+        """.format(table=self._table))
```

Comment:
format should not be used with SQL queries because it can lead to SQL injections. You should use instead %s replacement character and use pass a list parameters to the execute function `self.env.cr.execute(query, [self._table])`

Reply:
Ok, done ;)